test/functional/cmdLineTests/shareClassTests/SCCMLTests/ShareClassesCMLTests-3.xml

```diff
@@ -1621,7 +1621,322 @@
+		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+		<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+
+	<test id="Test 181: Test the sub-option createLayer" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ $currentMode$,createLayer -version</command>
+		<output type="success" caseSensitive="yes" regex="yes" javaUtilPattern="yes">(java|openjdk) version</output>
+
+		<output type="failure" caseSensitive="no" regex="no">Failed to start up the shared cache</output>
+		<output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+		<output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+		<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+
+	<test id="Test 182: Check whether the layer stats shows up correctly" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ $currentMode$,printstats</command>
+		<output type="success" caseSensitive="yes" regex="no">Current statistics for top layer of cache</output>
+		<output type="required" caseSensitive="yes" regex="yes" javaUtilPattern="yes">cache layer[\s]*= 1</output>
+		<output type="required" caseSensitive="yes" regex="no">Current statistics for all layers of cache</output>		
+		
+		<output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+		<output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+		<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+
+	<test id="Test 183: Test the sub-option Layer= with an incorrect layer number" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ $currentMode$,layer=3 -version</command>
+		<output type="success" caseSensitive="yes" regex="no">JVMSHRC844E Invalid layer number 3</output>
+		<output type="required" caseSensitive="yes" regex="no">The existing top layer is 1</output>
+		<output type="required" caseSensitive="yes" regex="no">The new layer number should be 2</output>
+
+		<output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+		<output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+		<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+
+	<exec command="rm -f testjavacore.txt" quiet="false"/>	
+
+	<!-- Generate the javacore file here in order to test it on test 185-->
+	<test id="Test 184: Test the sub-option Layer= with a correct layer number and generate the javacore file" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ -Xdump:java:events=vmstop,file=testjavacore.txt $currentMode$,layer=2 $CP_HANOI$ $PROGRAM_HANOI$</command>
+		<output type="success" caseSensitive="yes" regex="no">Puzzle solved!</output>
```

Comment:
We print out a message when creating the javacore file. Add a required condition to check that